sources/pmwebapi_test.go

```diff
@@ -55,49 +62,69 @@ func TestUnmarshal(t *testing.T) {
 	}
 }
 
-func TestLabelTypes(t *testing.T) {
-	units := ""
-	pcpType := ""
-	name := "kal"
-	response := typeLabel(units, pcpType, name)
-	expected := "kal"
-	if response != expected {
-		t.Fatalf("Failed when units and type strings were empty. Expected: %s, Got: %s", expected, response)
-	}
+func TestWeirdMetrics(t *testing.T) {
 
-	units = ""
-	pcpType = ""
-	name = ""
-	response = typeLabel(units, pcpType, name)
-	expected = ""
-	if response != expected {
-		t.Fatalf("Failed when all strings were empty. Expected: %s, Got: %s", expected, response)
+	name := make(map[string]string)
+	name["disk_total_nanosec"] = "disk_total_nanoseconds"
+	name["count / sec"] = "count_per_second"
+	name["Kbyte"] = "kilobytes"
+	name["nanosec"] = "nanoseconds"
+	name["millisec"] = "milliseconds"
+	name["failcnt"] = "failcount"
+	name["byte"] = "bytes"
+	name["Mbyte"] = "megabytes"
+	name["sec"] = "seconds"
+	name["bytes_byte"] = "bytes"
+	name["kilobytes"] = "kilobytes"
+	name["seconds"] = "seconds"
+	name["nanoseconds"] = "nanoseconds"
+	name["milliseconds"] = "milliseconds"
+	name["megabytes"] = "megabytes"
+	name["bytes"] = "bytes"
+	name["kal_byte"] = "kal_bytes"
+	name["gorgi_nanosec"] = "gorgi_nanoseconds"
+	name["asfdasfr_Mbyte"] = "asfdasfr_megabytes"
+	for key, val := range name {
+		converted := fixNaming(key)
+		if converted != val {
+			t.Fatalf("Failed when name was %s. Expected: %s, Got: %s", key, val, converted)
+		}
 	}
+}
 
-	units = "kal"
-	pcpType = "CoUnTeR"
-	name = "kal"
-	response = typeLabel(units, pcpType, name)
-	expected = "kal_kal_total"
-	if response != expected {
-		t.Fatalf("Failed when type was case sensitive. Expected: %s, Got: %s", expected, response)
-	}
+var tester []testlables
 
-	units = "tery"
-	pcpType = "gagage"
-	name = "kal"
-	response = typeLabel(units, pcpType, name)
-	expected = "kal_tery"
-	if response != expected {
-		t.Fatalf("Failed when type is any other than counter or empty. Expected: %s, Got: %s", expected, response)
+func TestLabelTypes(t *testing.T) {
+	tester = []testlables{
+		testlables{
+			units:    "",
+			tyype:    "",
+			name:     "kiaal",
+			expected: "kiaal",
+		},
+		testlables{
+			units:    "",
+			tyype:    "",
+			name:     "",
+			expected: "",
+		},
+		testlables{
+			units:    "kal",
+			tyype:    "CoUnTeR",
+			name:     "kal",
+			expected: "kal_kal_total",
+		},
+		testlables{
+			units:    "tery",
+			tyype:    "gagage",
+			name:     "kall",
+			expected: "kall_tery",
+		},
 	}
-
-	units = "87"
-	pcpType = "countER"
-	name = "k.a.l"
-	response = typeLabel(units, pcpType, name)
-	expected = "k_a_l_87_total"
-	if response != expected {
-		t.Fatalf("Failed when there were periods in the name. Expected: %s, Got: %s", expected, response)
+	for _ := range tester {
```

Comment:
I'm assuming you are going to fix this too?

Reply:
yup